overlay.d/05rhcos/usr/lib/dracut/modules.d/41rhcos-check-luks-syntax/rhcos-fail-boot-for-older-luks-syntax

```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+set -euo pipefail
+
+# This script tries to look for a `/etc/clevis.json` reference 
+# (used to enable LUKS in RHCOS 4.6 or earlier versions) in an 
+# Ignition config. If it exists and the version of RHCOS is 4.7, 
+# then the script ends with exit 1.
+
+ignition_cfg="/run/ignition.json"
+wanted_path="/etc/clevis.json"
+
+# select the `/etc/clevis.json` entry from a given Ignition config
+if jq -e ".storage.files[] | select(.path==\"${wanted_path}\")" "${ign_config}";  then
```

Comment:
If there is no `storage.files` section, this will still work correctly (return non-zero) but will produce a confusing error in the logs:

```
jq: error (at /run/ignition.json:1): Cannot iterate over null (null)
```

Reply:
ah, good catch. I think this should work: `.storage.files[]?`